src/main/java/com/meilisearch/sdk/SearchRequest.java

```diff
@@ -1,36 +1,40 @@
 package com.meilisearch.sdk;
 
 import com.meilisearch.sdk.model.MatchingStrategy;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
 import lombok.Setter;
 import lombok.experimental.Accessors;
 import org.json.JSONObject;
 
 /** Search request query string builder */
+@Builder
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+@NoArgsConstructor(access = AccessLevel.PACKAGE)
 @Getter
 @Setter
 @Accessors(chain = true)
 public class SearchRequest {
     private String q;
-    private int offset;
-    private int limit;
+    private Integer offset;
+    private Integer limit;
     private String[] attributesToRetrieve;
     private String[] attributesToCrop;
-    private int cropLength;
+    private Integer cropLength;
     private String cropMarker;
     private String highlightPreTag;
     private String highlightPostTag;
     private MatchingStrategy matchingStrategy;
     private String[] attributesToHighlight;
     private String[] filter;
     private String[][] filterArray;
-    private boolean showMatchesPosition;
+    private Boolean showMatchesPosition;
     private String[] facets;
     private String[] sort;
 
-    /** Empty SearchRequest constructor */
-    public SearchRequest() {}
```

Comment:
I think you can make it return `this`, and then you'll enable a new version (my preferred):

```java
new SearchRequest()
  .setQuery("my query") // or I think it will be just `setQ` - good as well.
  .setOffset(200)
  .setLimit(900);
```

WDYT?

Reply:
I modifying a test with this syntax `toStringQueryAndOffset`